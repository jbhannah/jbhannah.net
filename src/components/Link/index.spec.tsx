import { render, screen } from "@testing-library/react"
import * as React from "react"
import { Link } from "."

const TestLink = (props) => render(<Link {...props}>Test Link</Link>)

describe("Link", () => {
  describe("with an internal link", () => {
    beforeEach(() => TestLink({ href: "/" }))

    it("renders correctly", () => {
      expect(screen.getByText("Test Link")).toMatchSnapshot()
    })

    it("renders an a", () => {
      expect(screen.getByText("Test Link")).toHaveAttribute("href", "/")
    })

    it("does not add a target property", () => {
      expect(screen.getByText("Test Link")).not.toHaveAttribute("target")
    })

    it("does not add a rel property", () => {
      expect(screen.getByText("Test Link")).not.toHaveAttribute("rel")
    })
  })

  describe("with an anchor link", () => {
    beforeEach(() => TestLink({ href: "#fn-1" }))

    it("renders correctly", () => {
      expect(screen.getByText("Test Link")).toMatchSnapshot()
    })

    it("renders an a", () => {
      expect(screen.getByText("Test Link")).toHaveAttribute("href", "#fn-1")
    })

    it("does not add a target property", () => {
      expect(screen.getByText("Test Link")).not.toHaveAttribute("target")
    })

    it("does not add a rel property", () => {
      expect(screen.getByText("Test Link")).not.toHaveAttribute("rel")
    })
  })

  describe("with an external link", () => {
    beforeEach(() =>
      TestLink({
        href: "https://www.example.com",
      })
    )

    it("renders correctly", () => {
      expect(screen.getByText("Test Link")).toMatchSnapshot()
    })

    it("renders an a", () => {
      expect(screen.getByText("Test Link")).toHaveAttribute(
        "href",
        "https://www.example.com"
      )
    })

    it("adds a target property", () => {
      expect(screen.getByText("Test Link")).toHaveAttribute("target", "_blank")
    })

    it("adds a rel property", () => {
      expect(screen.getByText("Test Link")).toHaveAttribute("rel", "noopener")
    })
  })

  describe("with a rel property", () => {
    beforeEach(() =>
      TestLink({
        href: "https://www.example.com",
        rel: "license",
      })
    )

    it("renders correctly", () => {
      expect(screen.getByText("Test Link")).toMatchSnapshot()
    })

    it("appends to the rel property", () => {
      expect(screen.getByText("Test Link")).toHaveAttribute(
        "rel",
        "license noopener"
      )
    })
  })
})
