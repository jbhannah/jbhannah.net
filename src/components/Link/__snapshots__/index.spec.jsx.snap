// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link with an anchor link renders correctly 1`] = `
<a
  css={
    Object {
      "backgroundImage": "linear-gradient(to top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0) 1px, #0f7b3b 1px, #0f7b3b 2px, rgba(0, 0, 0, 0) 2px)",
      "borderRadius": "0.3em",
      "code": Object {
        "textShadow": "none",
      },
      "color": "#0f7b3b",
    }
  }
  href="#fn-1"
/>
`;

exports[`Link with an external link renders correctly 1`] = `
<a
  css={
    Object {
      "backgroundImage": "linear-gradient(to top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0) 1px, #0f7b3b 1px, #0f7b3b 2px, rgba(0, 0, 0, 0) 2px)",
      "borderRadius": "0.3em",
      "code": Object {
        "textShadow": "none",
      },
      "color": "#0f7b3b",
    }
  }
  href="https://www.example.com"
  onClick={[Function]}
  rel="noopener"
  target="_blank"
/>
`;

exports[`Link with an external link with a rel property renders correctly 1`] = `
<a
  css={
    Object {
      "backgroundImage": "linear-gradient(to top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0) 1px, #0f7b3b 1px, #0f7b3b 2px, rgba(0, 0, 0, 0) 2px)",
      "borderRadius": "0.3em",
      "code": Object {
        "textShadow": "none",
      },
      "color": "#0f7b3b",
    }
  }
  href="https://www.example.com"
  onClick={[Function]}
  rel="license noopener"
  target="_blank"
/>
`;

exports[`Link with an internal link renders correctly 1`] = `
<ForwardRef
  css={
    Object {
      "backgroundImage": "linear-gradient(to top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0) 1px, #0f7b3b 1px, #0f7b3b 2px, rgba(0, 0, 0, 0) 2px)",
      "borderRadius": "0.3em",
      "code": Object {
        "textShadow": "none",
      },
      "color": "#0f7b3b",
    }
  }
  to="/"
/>
`;
